pkg/artifactory/resource/security/resource_artifactory_trusted_key.go

```diff
@@ -0,0 +1,154 @@
+	ValidUntil  string `hcl:"valid_until" json:"valid_until"`
+}
+
+func ResourceArtifactoryTrustedKey() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: addTrustedKey,
+		DeleteContext: deleteTrustedKey,
+		ReadContext:   readTrustedKey,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "Manage the GPG trusted keys used to verify distributed release bundles",
+
+		Schema: map[string]*schema.Schema{
+			"key_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Generated by artifactory on upload",
+			},
+			"alias": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Will be used as a identifier when uploading/retrieving the public key via REST API",
+				ForceNew:    true,
+			},
+			"fingerprint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Generated by artifactory on upload",
+			},
+			"public_key": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The Public key to add as trusted distribution GPG key",
+				ForceNew:         true,
+				StateFunc:        stripTabs,
+				ValidateDiagFunc: validatePublicKey,
+			},
+			"issued_on": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date when this GPG key was created",
+			},
+			"issued_by": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name and eMail address of issuer",
+			},
+			"valid_until": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Date until this GPG key is valid",
+			},
+		},
+	}
+}
+
+func addTrustedKey(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	type keyPost struct {
+		Alias     string `json:"alias"`
+		PublicKey string `json:"public_key"`
+	}
+
+	var result trustedKeyPayLoad
+
+	resp, err := m.(*resty.Client).R().SetBody(keyPost{
+		d.Get("alias").(string),
+		strings.ReplaceAll(d.Get("public_key").(string), "\t", ""),
+	}).SetResult(&result).Post(trustedKeysAPIEndPoint)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if resp.StatusCode() != 200 && resp.StatusCode() != 201 {
+		return diag.FromErr(fmt.Errorf("unable to add key: http request failed: %s", resp.Status()))
+	}
+
+	d.Set("key_id", result.KeyID)
+	d.Set("fingerprint", result.Fingerprint)
+	d.Set("issued_on", result.IssuedOn)
+	d.Set("issued_by", result.IssuedBy)
+	d.Set("valid_until", result.ValidUntil)
+
+	d.SetId(result.KeyID)
+
+	return nil
+}
+
+func readTrustedKey(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+
+	type trustedKeysList struct {
+		Keys []trustedKeyPayLoad `json:"keys"`
+	}
+	data := trustedKeysList{}
+	resp, err := m.(*resty.Client).R().SetResult(&data).Get(trustedKeysAPIEndPoint)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if resp.StatusCode() != 200 {
```

Comment:
Use `resp.IsError()` instead.

Reply:
Also need to reset resource ID here. See https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource/security/resource_artifactory_permission_target.go#L346